cloud-config/master.yaml.tmpl

```diff
@@ -136,32 +137,145 @@ write_files:
+    spec:
+      # Number of Typha replicas.  To enable Typha, set this to a non-zero value *and* set the
+      # typha_service_name variable in the calico-config ConfigMap above.
+      #
+      # We recommend using Typha if you have more than 50 nodes.  Above 100 nodes it is essential
+      # (when using the Kubernetes datastore).  Use one replica for every 100-200 nodes.  In
+      # production, we recommend running at least 3 replicas to reduce the impact of rolling upgrade.
+      replicas: 0
+      revisionHistoryLimit: 2
+      template:
+        metadata:
+          labels:
+            k8s-app: calico-typha
+          annotations:
+            # This, along with the CriticalAddonsOnly toleration below, marks the pod as a critical
+            # add-on, ensuring it gets priority scheduling and that its resources are reserved
+            # if it ever gets evicted.
+            scheduler.alpha.kubernetes.io/critical-pod: ''
+        spec:
+          tolerations:
+          - key: CriticalAddonsOnly
+            operator: Exists
+          # Since Calico can't network a pod until Typha is up, we need to run Typha itself
+          # as a host-networked pod.
+          hostNetwork: true
+          serviceAccountName: calico-node
+          containers:
+          - image: quay.io/calico/typha:v0.6.0
+            name: calico-typha
+            ports:
+            - containerPort: 5473
+              name: calico-typha
+              protocol: TCP
+            env:
+              # Enable "info" logging by default.  Can be set to "debug" to increase verbosity.
+              - name: TYPHA_LOGSEVERITYSCREEN
+                value: "info"
+              # Disable logging to file and syslog since those don't make sense in Kubernetes.
+              - name: TYPHA_LOGFILEPATH
+                value: "none"
+              - name: TYPHA_LOGSEVERITYSYS
+                value: "none"
+              # Monitor the Kubernetes API to find the number of running instances and rebalance
+              # connections.
+              - name: TYPHA_CONNECTIONREBALANCINGMODE
+                value: "kubernetes"
+              - name: TYPHA_DATASTORETYPE
+                value: "kubernetes"
+              - name: TYPHA_HEALTHENABLED
+                value: "true"
+              # Uncomment these lines to enable prometheus metrics.  Since Typha is host-networked,
+              # this opens a port on the host, which may need to be secured.
+              #- name: TYPHA_PROMETHEUSMETRICSENABLED
+              #  value: "true"
+              #- name: TYPHA_PROMETHEUSMETRICSPORT
+              #  value: "9093"
+            livenessProbe:
+              httpGet:
+                path: /liveness
+                port: 9098
+              periodSeconds: 30
+              initialDelaySeconds: 30
+            readinessProbe:
+              httpGet:
+                path: /readiness
+                port: 9098
+              periodSeconds: 10
+
+    ---
+
```

Comment:
Do we use Typha in guest clusters?

Reply:
We still use old calico in Azure guest clusters and need to update manifests too. So Typha will be added.